olclient/bots.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+"""
+Classes and methods for bulk edits that match the following pattern:
+1. Search data dump for X condition
+2. Remove or modify record that fits condition X
+3. Create log of modification
+4. Repeat 1-3 for all records that meet condition X
+"""
+
+
+import argparse
+import datetime
+import logging
+import sys
+
+from olclient.openlibrary import OpenLibrary
+from os import makedirs, path
+
+
+class BaseBot(object):
+    def __init__(self, ol=None, dry_run=True, limit=1) -> None:
+        """Create logger and class variables"""
+        self.ol = ol or OpenLibrary()
+
+        self.parser = argparse.ArgumentParser(description=__doc__)
+        self.parser.add_argument('-f', '--file', type=str, default=None, help='Path to file containing input data')
+        self.parser.add_argument('-l, --limit', type=int, default=1,
+                                 help='Limit number of edits performed on external data.'
+                                      'Set to zero to allow unlimited edits.')
+        self.parser.add_argument('-d', '--dry-run', type=self._str2bool, default=True,
+                                 help='Execute the script without performing edits on external data.')
+        self.args = self.parser.parse_args()
+        self.dry_run = getattr(self.args, 'dry-run', None) or dry_run
+        self.limit = getattr(self.args, 'limit', None) or limit
+        self.changed = 0
+
+        job_name = path.splitext(sys.argv[0])[0]
+        self.logger = logging.getLogger("jobs.%s" % job_name)
+        self.logger.setLevel(logging.DEBUG)
+        log_formatter = logging.Formatter('%(name)s;%(levelname)-8s;%(asctime)s %(message)s')
+        self.console_handler = logging.StreamHandler()
+        self.console_handler.setLevel(logging.WARN)
+        self.console_handler.setFormatter(log_formatter)
+        self.logger.addHandler(self.console_handler)
+        here = path.dirname(path.abspath(__file__))
+        log_dir = ''.join([here, '/logs/jobs/', job_name])
```

Comment:
```suggestion
        home = path.expanduser("~")
        log_dir = ''.join([home, '/logs/jobs/', job_name])
```

Reply:
https://docs.python.org/3/library/os.path.html#os.path.expanduser

Reply:
I believe that `$HOME` should always be `/home/travis/` on Travis CI

Reply:
https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

Reply:
No longer applicable